File: 868-binary-gap.java
class Solution {
    public int binaryGap(int n) {
        String val = Integer.toBinaryString(n);
        int index = -1;
        int ans = 0;
        for(int i=0;i<val.length();i++){
            if(val.charAt(i) == '1'){
                if(index != -1){
                    int temp = i - index;
                    ans = temp > ans ? temp: ans;
                }
                index = i;
            }
        }
        return ans;
    }
}

File: 868-binary-gap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        val = bin(n)[2:]
        index = -1
        ans = 0
        for i, num in enumerate(val):
            if num == '1':
                if index != -1:
                    ans = max(i - index,ans)
                index = i
        return ans
